v0/destinations/branch/transform.js

```diff
@@ -83,25 +63,48 @@ function getCategoryAndName(rudderEventName) {
 function getUserData(message) {
   const { context } = message;
   const { os } = message.context;
-
   if (!os || !isDefinedAndNotNullAndNotEmpty(os.name)) {
     throw new CustomError(
       "os name is missing in the payload and please make sure to insert it at context.os.name",
       400
     );
   }
+  const userData = {};
+  userData.os = os.name;
+  userData.os_version = os.version;
+  userData.app_version = context.app.version;
+  userData.screen_dpi = context.screen.density;
+  userData.developer_identity = getFieldValueFromMessage(message, "userId");
 
-  return removeUndefinedAndNullValues({
-    os: os.name,
-    os_version: os.version,
-    app_version: context.app.version,
-    screen_dpi: context.screen.density,
-    android_id: get(context, "android_id") ? context.android_id : null,
-    idfa: get(context, "idfa") ? context.android_id : null,
-    idfv: get(context, "idfv") ? context.android_id : null,
-    aaid: get(context, "aaid") ? context.android_id : null,
-    developer_identity: getFieldValueFromMessage(message, "userId")
-  });
+  if (isAppleFamily(os.name)) {
+    userData.idfa = get(message, "context.device.advertisingId");
+    userData.idfv = get(message, "context.device.id");
+  } else if (os.name.toLowerCase() === "android") {
+    userData.android_id = get(message, "context.device.id");
+    userData.aaid = get(message, "context.device.advertisingId");
+  }
+
+  const att = get(message, "context.device.attTrackingStatus");
+  if (isDefinedAndNotNull(att)) {
+    if (att === 3) {
+      userData.limit_ad_tracking = false;
+    } else if (att === 2) {
+      userData.limit_ad_tracking = true;
+    }
+  }
+
+  if (
+    userData.developer_identity ||
+    (userData.os === "android" && userData.aaid) ||
+    userData.android_id ||
+    (userData.os === "ios" && userData.idfa) ||
+    userData.idfv
+  )
+    return removeUndefinedAndNullValues(userData);
+  throw new CustomError(
+    "developer_identity or (os=iOS AND (idfa OR idfv)) or (os=Android AND (android_id or aaid)) is required in user_data.",
```

Comment:
This error message only makes sense if `Branch` is throwing it. We need to make it clear to the user in terms of what's missing from the input event.

Reply:
Okay, so to do that we have to check what is missing, I think showing them the specification will be enough for them to figure out what they are missing.

Reply:
```suggestion
    "At least one of the user ID, advertising ID (device.advertisingId), or device ID (device.id) is missing.",
```

Reply:
Got it